### PR TITLE
[SPARK-5601][MLLIB] make streaming linear algorithms Java-friendly

### PR DESCRIPTION
**Pull request by mengxr**
Overload `trainOn`, `predictOn`, and `predictOnValues`.

CC @freeman-lab

### CODE REVIEW

File: mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+package org.apache.spark.mllib.regression;
+
+import java.io.Serializable;
+import java.util.List;
+
+import scala.Tuple2;
+
+import com.google.common.collect.Lists;
+import static org.apache.spark.streaming.JavaTestUtils.*;
```
Comment (jkbradley) [inline mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java]: order

**Comment by jkbradley** [comment]
Looks good, but not too familiar with this class

File: mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    List<Tuple2<Integer, Vector>> testBatch = Lists.newArrayList(
+      new Tuple2<Integer, Vector>(10, Vectors.dense(1.0)),
+      new Tuple2<Integer, Vector>(11, Vectors.dense(0.0)));
+    JavaPairDStream<Integer, Vector> test = JavaPairDStream.fromJavaDStream(
+      attachTestInputStream(ssc, Lists.newArrayList(testBatch, testBatch), 2));
+    StreamingLinearRegressionWithSGD slr = new StreamingLinearRegressionWithSGD()
+      .setNumIterations(2)
+      .setInitialWeights(Vectors.dense(0.0));
+    slr.trainOn(training);
+    JavaPairDStream<Integer, Double> prediction = slr.predictOnValues(test);
```
Comment (freeman-lab) [inline mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java]: Worth adding a basic test for accuracy, or maybe that's unnecessary duplication?

File: mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/*
```
Comment (freeman-lab) [inline mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java]: Do you want to add a similar `JavaStreamingLogisticRegressionSuite`? Given that the primary change here is to the base class (used by both linear and logistic regression), might make sense to include in the same PR.

**Comment by freeman-lab** [comment]
@mengxr nice patch! I left two pretty minor comments about expanding the tests, otherwise looking good.

File: mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java (hunk 2 above)
Comment (mengxr) [inline mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java]: That test is in the Scala suite. This is just to test Java APIs. It is harder to maintain if we check accuracy in both places.

Comment (freeman-lab) [inline mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java]: Cool, agreed.

File: mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java (hunk 3 above)
Comment (mengxr) [inline mllib/src/test/java/org/apache/spark/mllib/regression/JavaStreamingLinearRegressionSuite.java]: Sure.

**Comment by mengxr** [comment]
Merged into master and branch-1.3.

**Pull request closed without merging**